sbt-version-policy/src/main/scala/sbtversionpolicy/DependencyCheckReport.scala

```diff
@@ -153,24 +155,37 @@ object DependencyCheckReport {
         previousVersion == currentVersion
       case VersionCompatibility.SemVer | VersionCompatibility.EarlySemVer | VersionCompatibility.SemVerSpec =>
         // Early SemVer and SemVer Spec are equivalent regarding source compatibility
-        extractSemVerNumbers(currentVersion).zip(extractSemVerNumbers(previousVersion)).headOption match {
-          case Some(((currentMajor, currentMinor, currentPatch), (previousMajor, previousMinor, previousPatch))) =>
-            currentMajor == previousMajor && {
-              if (currentMajor == 0)
-                currentMinor == previousMinor && currentPatch == previousPatch
-              else
-                currentMinor == previousMinor && currentPatch >= previousPatch
+        (extractSemVerNumbers(currentVersion), extractSemVerNumbers(previousVersion)) match {
+          case (Some(currentSemVer), Some(previousSemVer)) =>
+            def sameMajor  = currentSemVer.major == previousSemVer.major
+            def sameMinor  = currentSemVer.minor == previousSemVer.minor
+            def samePatch  = currentSemVer.patch == previousSemVer.patch
+            def sameSuffix = currentSemVer.suffix == previousSemVer.suffix
+
+            if (currentSemVer.major == 0) {
+              sameMajor && sameMinor && samePatch && sameSuffix
+            } else {
+              // 1.0.0-RC1 may be source incompatible to 1.0.0-RC2
+              // but!
+              // 1.0.1-RC2 must be source compatible both to 1.0.1-RC1 and 1.0.0 (w/o suffix!)
+              def compatPatch = 
+                (currentSemVer.patch > 0 && samePatch) || 
+                (currentSemVer.patch > previousSemVer.patch && previousSemVer.suffix.isEmpty)
+
+              sameMajor && sameMinor && ((samePatch && sameSuffix) || compatPatch)
```

Comment:
I wonder if this could be simplified to:

```suggestion
              def compatPatch =
                previousSemVer.suffix.isEmpty || previousSemVer.patch > 0

              sameMajor && sameMinor && compatPatch
```

(untested)

Reply:
I've used your check, but we still need `(samePatch && sameSuffix)` to allow `1.0.0-RC1` be source compat to itself. :(